storage: increase the load-based splitting QPS threshold from 250 to 2500

ajwerner opened the pull request:
Recent improvements in write efficiency and batching lead to a re-evaluation
of the reasons for load-based splitting. Intuitively fewer splits ought to
offer increased batching opportunities while more splits ought to offer
increased concurrency. Above a given number it is not obvious that
increased concurrency will translate effectively to increased parallelism.

Experimental evidence shows that the right threshold for load-based splitting
is now closer to 2500 than 250. It also shows that over-splitting can have
negative effects on latency and throughput.

Load-based splitting remains important additionally for the opportunity it
provides to balance load. Load-balancing however is not currently a part of the
splitting heuristic.

The second commit in the PR adds roachtests which do not perform any manual splits. 

Release note (performance improvement): Adjust load-based splitting QPS
threshold to avoid over-splitting.

Commit:
storage: increase the load-based splitting QPS threshold from 250 to 2500

Recent improvements in write efficiency and batching lead to a reevaluation
of the reasons for load-based splitting. Intuitively fewer splits ought to
offer increased batching opportunities while more splits ought to offer
increased concurrency. Above a given number it is not obvious that
increased concurrency will translate effectively to increased parallelism.

Experimental evidence shows that the right threshold for load-based splitting
is now closer to 2500 than 250. It also shows that over-splitting can have
negative effects on latency and throughput.

Load-based splitting remains important additionally for the opportunity it
provides to balance load. Load-balancing however is not currently a part of the
splitting heuristic.

Release note (performance improvement): Adjust load-based splitting QPS
threshold to avoid over-splitting.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/39687)
<!-- Reviewable:end -->

Commit:
roachtest: add KV configurations with no manual splits

Before this PR we always ran our KV benchmarks with 1000 manual splits.
This is almost certainly too many. We preserve this value for the historical
continuity of the tests but add new configurations which do no manual
splitting.

Release note: None

ajwerner commented:
TFTR!

I ran a quick benchmark of kv0 with different QPS thresholds and here's what I got:

```
name                                                           ops/s
Cockroach-concurrency=1024-qps_thresh=250-reads=0-throughput   18.1k ± 1%
Cockroach-concurrency=1024-qps_thresh=2500-reads=0-throughput  20.7k ± 2%
Cockroach-concurrency=1024-qps_thresh=5000-reads=0-throughput  19.9k ± 1%
```

Seems to support that we've picked a good number. 

bors r+
